tls/s2n_crl.c

```diff
@@ -91,6 +91,47 @@ int s2n_crl_get_issuer_hash(struct s2n_crl *crl, uint64_t *hash)
     return S2N_SUCCESS;
 }
 
+int s2n_crl_validate_this_update(struct s2n_crl *crl)
+{
+    POSIX_ENSURE_REF(crl);
+    POSIX_ENSURE_REF(crl->crl);
+
+    ASN1_TIME *this_update = X509_CRL_get_lastUpdate(crl->crl);
+    POSIX_ENSURE_REF(this_update);
+
+    int ret = X509_cmp_time(this_update, NULL);
+    if (ret == 0) {
+        POSIX_BAIL(S2N_ERR_CRL_INVALID_THIS_UPDATE);
+    }
+    if (ret > 0) {
+        POSIX_BAIL(S2N_ERR_CRL_NOT_YET_VALID);
+    }
+
+    return S2N_SUCCESS;
+}
+
+int s2n_crl_validate_next_update(struct s2n_crl *crl)
+{
+    POSIX_ENSURE_REF(crl);
+    POSIX_ENSURE_REF(crl->crl);
+
+    ASN1_TIME *next_update = X509_CRL_get_nextUpdate(crl->crl);
+    if (next_update == NULL) {
+        /* If the CRL has no nextUpdate field, assume it will never expire */
+        return S2N_SUCCESS;
+    }
+
+    int ret = X509_cmp_time(next_update, NULL);
+    if (ret == 0) {
+        POSIX_BAIL(S2N_ERR_CRL_INVALID_NEXT_UPDATE);
+    }
+    if (ret < 0) {
+        POSIX_BAIL(S2N_ERR_CRL_EXPIRED);
+    }
```

Comment:
Might be cleaner?
```suggestion
    POSIX_ENSURE(ret != 0, S2N_ERR_CRL_INVALID_NEXT_UPDATE);
    POSIX_ENSURE(ret > 0, S2N_ERR_CRL_EXPIRED);
```

Reply:
Yes, that looks better! 5ed638cf6f2d99fef6edbf0d60785c214b826743